// @line/bot-sdk/dist/cjs/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineBotClient = exports.webhook = exports.shop = exports.moduleAttach = exports.moduleOperation = exports.messagingApi = exports.manageAudience = exports.liff = exports.insight = exports.channelAccessToken = exports.validateSignature = exports.middleware = void 0;
const middleware_js_1 = __importDefault(require("./middleware.js"));
exports.middleware = middleware_js_1.default;
const validate_signature_js_1 = __importDefault(require("./validate-signature.js"));
exports.validateSignature = validate_signature_js_1.default;
// re-export exceptions and types
__exportStar(require("./exceptions.js"), exports);
__exportStar(require("./types.js"), exports);
const channelAccessToken = __importStar(require("./channel-access-token/api.js"));
exports.channelAccessToken = channelAccessToken;
const insight = __importStar(require("./insight/api.js"));
exports.insight = insight;
const liff = __importStar(require("./liff/api.js"));
exports.liff = liff;
const manageAudience = __importStar(require("./manage-audience/api.js"));
exports.manageAudience = manageAudience;
const messagingApi = __importStar(require("./messaging-api/api.js"));
exports.messagingApi = messagingApi;
// Note: `module` is reserved word in Javascript.
const moduleOperation = __importStar(require("./module/api.js"));
exports.moduleOperation = moduleOperation;
const moduleAttach = __importStar(require("./module-attach/api.js"));
exports.moduleAttach = moduleAttach;
const shop = __importStar(require("./shop/api.js"));
exports.shop = shop;
const webhook = __importStar(require("./webhook/api.js"));
exports.webhook = webhook;
var line_bot_client_js_1 = require("./line-bot-client.js");
Object.defineProperty(exports, "LineBotClient", { enumerable: true, get: function () { return line_bot_client_js_1.LineBotClient; } });
