// @line/bot-sdk/dist/cjs/line-bot-client.factory.generated.js
"use strict";
/**
 * This file is autogenerated.
 *
 * Generated by scripts/line-bot-client-generator/main.mjs
 * Do not edit this file directly.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.createLineBotClientDelegates = createLineBotClientDelegates;
const insight = __importStar(require("./insight/api.js"));
const liff = __importStar(require("./liff/api.js"));
const manageAudience = __importStar(require("./manage-audience/api.js"));
const messagingApi = __importStar(require("./messaging-api/api.js"));
const moduleOperation = __importStar(require("./module/api.js"));
const moduleAttach = __importStar(require("./module-attach/api.js"));
const shop = __importStar(require("./shop/api.js"));
function createLineBotClientDelegates(config) {
    return {
        insight: new insight.InsightClient({
            baseURL: config.apiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        liff: new liff.LiffClient({
            baseURL: config.apiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        manageAudience: new manageAudience.ManageAudienceClient({
            baseURL: config.apiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        manageAudienceBlob: new manageAudience.ManageAudienceBlobClient({
            baseURL: config.dataApiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        messagingApi: new messagingApi.MessagingApiClient({
            baseURL: config.apiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        messagingApiBlob: new messagingApi.MessagingApiBlobClient({
            baseURL: config.dataApiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        lineModule: new moduleOperation.LineModuleClient({
            baseURL: config.apiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        lineModuleAttach: new moduleAttach.LineModuleAttachClient({
            baseURL: config.managerBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
        shop: new shop.ShopClient({
            baseURL: config.apiBaseURL,
            channelAccessToken: config.channelAccessToken,
            defaultHeaders: config.defaultHeaders,
        }),
    };
}
